tracer.go

```diff
@@ -245,7 +245,10 @@ func (t *Tracer) startSpanWithOptions(
 			continue
 		}
 
-		references = append(references, Reference{Type: ref.Type, Context: ctxRef})
+		if ctxRef.IsValid() {
```

Comment:
Perhaps we should rename `isValidReferece` on L238 to sth else? The logic is a bit confusing to me

Reply:
do you have suggestions? `isThereSomethingThere(ref)`?

Reply:
or `couldBeParent(ref)`

Reply:
or we can simply inline the condition, since it's never reused

Reply:
I think inlining and refactoring the logic for the debug container closer would help. 

My confusion was  because `isValidReference(ctxRef)` already evaluates `ctxRef.isValid()` on L238 and short circuits. Upon first reading I expected L248 to always be valid.

Reply:
it's reused so cannot inline it. I renamed it to isEmptyReference